tests/unittests/test_mdal_utils.cpp

```diff
@@ -221,3 +221,46 @@ TEST( MdalUtilsTest, EndsWidth )
     EXPECT_EQ( test.second, MDAL::endsWith( test.first, "cd", MDAL::ContainsBehaviour::CaseInsensitive ) );
   }
 }
+
+struct LoadMeshUri
+{
+  LoadMeshUri( std::string u, std::string d, std::string mf, std::string mn, int mid ) :
+    uri( u ),
+    expectedDriver( d ),
+    expectedMeshFile( mf ),
+    expectedMeshName( mn ),
+    expectedMeshID( mid ) {};
+
+  std::string uri;
+  std::string expectedDriver;
+  std::string expectedMeshFile;
+  std::string expectedMeshName;
+  int expectedMeshID;
+};
+
+TEST( MdalUtilsTest, ParseLoadMeshUri )
+{
+  std::string inputUri, driverName, meshFile, specificMeshName;
+  int specificMeshID;
+
+  std::vector<LoadMeshUri> tests
+  {
+    LoadMeshUri( "Ugrid:\"mesh.nc\":mesh1d", "Ugrid", "mesh.nc", "mesh1d", 0 ),
```

Comment:
add test for mesh name with whitespaces "Ugrid:\"C:\\myfile\.  \\with spaces\\hi.nc":\"incredible mesh\""

Reply:
sure, good idea. However, I think that Ugrid convention can not have whitespaces in mesh name